packages/opentelemetry-sdk-trace-base/test/common/BasicTracerProvider.test.ts

```diff
@@ -240,40 +265,99 @@ describe('BasicTracerProvider', () => {
     });
   });
 
-  describe('.register()', () => {
-    describe('propagator', () => {
-      class DummyPropagator implements TextMapPropagator {
-        inject(
-          context: Context,
-          carrier: any,
-          setter: TextMapSetter<any>
-        ): void {
-          throw new Error('Method not implemented.');
-        }
-        extract(
-          context: Context,
-          carrier: any,
-          getter: TextMapGetter<any>
-        ): Context {
-          throw new Error('Method not implemented.');
+  describe('Custom TracerProvider through inheritance', () => {
+    beforeEach(() => {
+      envSource.OTEL_TRACES_EXPORTER = 'custom-exporter';
+      envSource.OTEL_PROPAGATORS = 'custom-propagator';
+    });
+
+    afterEach(() => {
+      delete envSource.OTEL_TRACES_EXPORTER;
+      delete envSource.OTEL_PROPAGATORS;
+      sinon.restore();
+    });
+
+    it('can be extended by overriding registered components', () => {
+      class CustomTracerProvider extends BasicTracerProvider {
+        protected static override readonly _registeredPropagators = new Map<
+          string,
+          () => TextMapPropagator
+            >([
+              ['custom-propagator', () => new DummyPropagator()],
+            ]);
+
+        protected static override readonly _registeredExporters = new Map<
+          string,
+          () => SpanExporter
+            >([
+              ['custom-exporter', () => new DummyExporter()],
+            ]);
+      }
```

Comment:
Just to make sure I understand - since we override `_registeredPropagators` and `_registeredExporters` from base class, now the only possible propagators/exporters are those that are set here, right? so if `OTEL_PROPAGATORS` contains: `[tracecontext]`, that will not work.

To make extending class use both parent and child propagators will require child class to use:
```js
  protected static override readonly _registeredPropagators = new Map<
    string,
    PROPAGATOR_FACTORY
  >([
    ...BasicTracerProvider._registeredPropagators,
    [
      'foo',
      () =>
        new FooPropagator(),
    ],
```

Reply:
That's correct. So instead overriding the static list AND the getter, one can get away with doing just one: the static list.